src/api/services.js

```diff
@@ -1,109 +1,119 @@
-const api = require('./index');
-const addr = require('../env_config').default;
+const api = require("./index");
+const addr = require("../env_config").default;
 const sessionStorage = window.sessionStorage;
 const token = sessionStorage.getItem("token");
 
-exports.login =  function(data, callback){
-    api.post(addr+"/auth/login/",
-    {"content-type": "application/json"},
-    data, function(error, token){
-        //callback of the method here
-        console.log("error", error);
-         if(error){
-            console.log(error);
-            return callback(error);
-         }else{
-            return callback(null, token.auth_token);
-         }
-        });
-}
-
-exports.addChew =  function(data, callback){
-   api.post(addr+"/api/v1/chews",
-   {"content-type": "application/json",
-   "Authorization": "Token "+token
-},
-   data, function(error, response){
-       //callback of the method here
-       console.log("error", error);
-        if(error){
-           console.log(error);
-           return callback(error);
-        }else{
-           return callback(null, response);
-        }
-       });
-}
-exports.addMidwife =  function(data, callback){
-   api.post(addr+"/api/v1/midwives",
-   {"content-type": "application/json",
-   "Authorization": "Token "+token
-},
-   data, function(error, response){
-       //callback of the method here
-       console.log("error", error);
-        if(error){
-           console.log(error);
-           return callback(error);
-        }else{
-           return callback(null, response);
-        }
-       });
-}
-exports.addAmbulance =  function(data, callback){
-   api.post(addr+"/api/v1/ambulances",
-   {"content-type": "application/json",
-   "Authorization": "Token "+token
-},
-   data, function(error, response){
-       //callback of the method here
-       console.log("error", error);
-        if(error){
-           console.log(error);
-           return callback(error);
-        }else{
-           return callback(null, response);
-        }
-       });
-}
+exports.login = function(data, callback) {
+  api.post(
+    addr + "/auth/login/",
+    { "content-type": "application/json" },
+    data,
+    function(error, response) {
+      if (error) {
+        return callback(error);
+      } else {
+        let district = response.user && response.user.village && response.user.village.parish && response.user.village.parish.sub_county && response.user.village.parish.sub_county.county && response.user.village.parish.sub_county.county.district && response.user.village.parish.sub_county.county.district;
```

Comment:
Feels alittle complicated and counter intuitive. You are checking if you have a user, and if the user is associated to a village, parish, subcounty etc. 

Just check if you have response.user && response.user.village.parish.sub_county.county.district;

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
I have read about optional chaining, but i requires a create react app upgrade, or add two external libraries to enable it.
https://medium.com/better-programming/how-to-use-the-optional-chaining-operator-in-your-javascript-app-a08f28db13e9